payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayLauncher.kt

```diff
@@ -62,29 +64,20 @@ class GooglePayLauncher internal constructor(
         readyCallback: ReadyCallback,
         resultCallback: ResultCallback
     ) : this(
-        activity.lifecycleScope,
-        config,
-        readyCallback,
-        activity.registerForActivityResult(
-            GooglePayLauncherContract()
-        ) {
-            resultCallback.onResult(it)
-        },
-        googlePayRepositoryFactory = {
-            DefaultGooglePayRepository(
-                context = activity.application,
-                environment = config.environment,
-                billingAddressParameters = config.billingAddressConfig.convert(),
-                existingPaymentMethodRequired = config.existingPaymentMethodRequired,
-                allowCreditCards = config.allowCreditCards
-            )
-        },
-        PaymentAnalyticsRequestFactory(
+        applicationContext = activity.applicationContext,
+        lifecycleScope = activity.lifecycleScope,
+        config = config,
+        readyCallback = readyCallback,
+        launcher = activity.registerForActivityResult(
+            GooglePayLauncherContract(),
+            resultCallback::onResult,
+        ),
+        analyticsRequestFactory = PaymentAnalyticsRequestFactory(
```

Comment:
There are a lot of changes in this PR that no longer seem relevant after using ErrorReporter.createFallbackInstance.

Can most of this be reverted?

I think it's like 80% of the PR, if that's the case.